libfreerdp/core/nla.c

```diff
@@ -619,40 +814,231 @@ static int nla_client_init(rdpNla* nla)
 			        settings->RedirectionPasswordLength / sizeof(WCHAR) - 1) < 0)
 				return -1;
 		}
-		else
-		{
-			BOOL usePassword = TRUE;
 
-			if (settings->RestrictedAdminModeRequired)
+		if (settings->RestrictedAdminModeRequired)
+		{
+			if (settings->PasswordHash && strlen(settings->PasswordHash) == 32)
 			{
-				if (settings->PasswordHash)
-				{
-					if (strlen(settings->PasswordHash) == 32)
-					{
-						if (sspi_SetAuthIdentity(nla->identity, settings->Username,
-						                         settings->Domain, settings->PasswordHash) < 0)
-							return -1;
-
-						/**
-						 * Increase password hash length by LB_PASSWORD_MAX_LENGTH to obtain a
-						 * length exceeding the maximum (LB_PASSWORD_MAX_LENGTH) and use it this for
-						 * hash identification in WinPR.
-						 */
-						nla->identity->PasswordLength += LB_PASSWORD_MAX_LENGTH;
-						usePassword = FALSE;
-					}
-				}
+				if (sspi_SetAuthIdentity(nla->identity, settings->Username, settings->Domain,
+				                         settings->PasswordHash) < 0)
+					return -1;
+
+				/**
+				 * Increase password hash length by LB_PASSWORD_MAX_LENGTH to obtain a
+				 * length exceeding the maximum (LB_PASSWORD_MAX_LENGTH) and use it this for
+				 * hash identification in WinPR.
+				 */
+				nla->identity->PasswordLength += LB_PASSWORD_MAX_LENGTH;
+				usePassword = FALSE;
 			}
+		}
+
+		if (usePassword)
+		{
+			if (sspi_SetAuthIdentity(nla->identity, settings->Username, settings->Domain,
+			                         settings->Password) < 0)
+				return -1;
+		}
+	}
 
-			if (usePassword)
+	return TRUE;
+}
+
+#ifdef WITH_GSSAPI
+static const char* parseInt(const char* v, INT32* r)
```

Comment:
Why not use `strtol` and similar?

Reply:
To not have all the complicated behaviours and drawbacks of `strtol`.